test.c

```diff
@@ -4,18 +4,25 @@ int main()
 {
     XMLDocument doc;
     if (XMLDocument_load(&doc, "test.xml")) {
+        char fields_buff[1024] = { 0 };
         XMLNode* str = XMLNode_child(doc.root, 0);
 
         XMLNodeList* fields = XMLNode_children(str, "field");
         for (int i = 0; i < fields->size; i++) {
             XMLNode* field = XMLNodeList_at(fields, i);
             XMLAttribute* type = XMLNode_attr(field, "type");
-            type->value = "";
+            snprintf(fields_buff, 1024, "%s %s\n", type->key, type->value);
```

Comment:
This was reassigning a `heap-allocated` address space which was not free-able and would raise a memory access violation. I think it's useful to provide an example of how the structs can be used so have just provided an alternate behaviour implementation.